### ISSUE TITLE
$HOSTNAME not resolving as environment variable as referenced in record_modifier

### ISSUE THREAD

**Issue by jbhilton11** [fluent/fluent-bit#460]
I'm trying to add the hostname of the server the forwarder is running on. In the record_modifier example, it shows adding hostname as ${HOSTNAME} which should resolve (as an already-established) environment variable to the hostname of the server. This, however, doesn't resolve to anything and therefore the field is left out of the record.

**Comment by edsiper** [comment]
would you please share your configuration file ?

**Comment by jbhilton11** [comment]
Here is most of my config below. I do not use a SET variable for HOSTNAME because in the example for the [record_modifier](http://fluentbit.io/documentation/current/filter/record_modifier.html#append-fields), it does not have HOSTNAME defined as a SET varaible. I figured it'd be available since it's always defined as an environment variable. I have verified that the HOSTNAME environment variable equates to what my hostname should be by running: `echo $HOSTNAME` on my system.
 
```
<Defined @SET variables used below>

[INPUT]
  Name tail
  Tag varLogAudit
  Path /var/log/audit/audit.log
  Parser auditd
  DB <path to file>/fluent-bit-state.db

[INPUT]
  Name proc
  Tag heartbeat
  Proc_Name fluent-bit
  Interval_Sec 60
  Fd false
  Mem false

[FILTER]
  Name record_modifier
  Match *
  ... (adding other record fields)
  Record hostname ${HOSTNAME}

[FILTER]
  Name record_modifier
  Match heartbeat
  ... (adding other record fields)
  Remove_key proc_name

[OUTPUT]
  Name http
  Match *
  Host ****
  Port ****
  URI ****
  Format json
  Retry_limit False
  tls On
  ... (other tls options)
```

**Comment by jbhilton11** [comment]
I was curious if there has been any further thought on this issue?

**Comment by edsiper** [comment]
@jbhilton11 the pre-set of HOSTNAME was fixed on version v0.12.6, here are the release notes:

   http://fluentbit.io/announcements/v0.12.6/

if you are using a newer version and you face the same issue please let me know

**Comment by edsiper** [comment]
Fixed.
